web-ui/src/components/menu/Menu.jsx

```diff
@@ -200,7 +201,16 @@ function Menu() {
           aria-haspopup="true"
           onClick={handleToggle}
         >
-          <Link style={{ textDecoration: "none" }} to="/profile">
+          <Link
+            onClick={() =>
+              dispatch({
+                type: SET_SELECTED_MEMBER,
+                payload: userProfile.memberProfile,
+              })
+            }
```

Comment:
This dispatch shouldn't be necessary. The change in the URL should drive this.

Reply:
I'm a bit confused by all the selected user stuff. I think it's likely unnecessary. Grab a few minutes with me and let's hash it out.